### PR TITLE
OCPQE-14239: Integrate jq in the image to avoid download jq from internet

### PR DESCRIPTION
**Pull request by liangxia**
Changes in this PR,
1. Switch pod image from ansible to verification-tests, as the later one has jq in it.
2. Remove the codes that download jq from github, as interact with external network is unstable.
3. Use proper openshift-ansible branch and ansible version based on OCP version.

### CODE REVIEW

**Comment by liangxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-stable-4.13-upgrade-from-stable-4.12-azure-upi-p3-f28

**Comment by liangxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-arm64-nightly-4.12-upgrade-from-stable-4.11-aws-ipi-ovn-p2-f14

File: ci-operator/step-registry/cucushift/upgrade/toimage/cucushift-upgrade-toimage-ref.yaml [hunk 1]
```diff
@@ -1,7 +1,10 @@
  ref:
   as: cucushift-upgrade-toimage
-  from: ansible
-  timeout: 24h
+  from_image:
```
Comment (jhou1) [inline ci-operator/step-registry/cucushift/upgrade/toimage/cucushift-upgrade-toimage-ref.yaml:3]: @shellyyang1989 Could you review?

**Comment by shellyyang1989** [comment]
The change looks good to me. It would be great to run a RHEL upgrade to test `openshift-ansible` tool in verification-tests works well.

**Comment by liangxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-4.13-upgrade-from-stable-4.12-azure-ipi-workers-rhel8-p2-f14

**Comment by JianLi-RH** [comment]
will this PR fix https://issues.redhat.com/browse/OCPQE-14239?

**Comment by liangxia** [comment]
> The change looks good to me. It would be great to run a RHEL upgrade to test `openshift-ansible` tool in verification-tests works well.

@shellyyang1989 Pass [log](https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/38103/rehearse-38103-periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-4.13-upgrade-from-stable-4.12-azure-ipi-workers-rhel8-p2-f14/1644164428502929408/artifacts/azure-ipi-workers-rhel8-p2-f14/workers-rhel-scaleup/build-log.txt)

**Comment by shellyyang1989** [comment]
> > The change looks good to me. It would be great to run a RHEL upgrade to test `openshift-ansible` tool in verification-tests works well.
> 
> @shellyyang1989 Pass [log](https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/38103/rehearse-38103-periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-4.13-upgrade-from-stable-4.12-azure-ipi-workers-rhel8-p2-f14/1644164428502929408/artifacts/azure-ipi-workers-rhel8-p2-f14/workers-rhel-scaleup/build-log.txt)

Thank you Liang! Since the change is being made on upgrade-toimage step, we need to check the result of upgrade. Looks like [upgrade](https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/38103/rehearse-38103-periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-4.13-upgrade-from-stable-4.12-azure-ipi-workers-rhel8-p2-f14/1644164428502929408/artifacts/azure-ipi-workers-rhel8-p2-f14/cucushift-upgrade-toimage/build-log.txt) was stuck on the node:

```
06:07:53 +0000   KubeletNotReady              container runtime status check may not have completed yet
```
If possible, let's re-run the test or select another workflow. Wdyt?

**Comment by liangxia** [comment]
> If possible, let's re-run the test or select another workflow. Wdyt?

Upgrade-to-image step has been verified by the other two tests.

**Comment by shellyyang1989** [comment]
> will this PR fix https://issues.redhat.com/browse/OCPQE-14239?

Not exactly. It will fix part of the issue in OCPQE-14239. As for OCPQE-14239, @dis016 has submitted a PR (Thank you) which is  pending on issue https://github.com/openshift/release/pull/37250#issuecomment-1491220810. Will you take over it?

**Comment by shellyyang1989** [comment]
> > If possible, let's re-run the test or select another workflow. Wdyt?
> 
> Upgrade-to-image step has been verified by the other two tests.

You're right that RHCOS upgrade can be verified by those tests. In addition, upgrade-toimage includes RHEL node upgrade as well which uses openshit-ansible tool in previous ansible image. As you're changing the image, so we need to make sure it will not introduce regression on RHEL upgrade as well.

**Comment by liangxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-4.13-upgrade-from-stable-4.12-azure-ipi-workers-rhel8-p2-f14

**Comment by liangxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.11-amd64-nightly-4.11-upgrade-from-stable-4.10-aws-ipi-proxy-workers-rhel8-p2-f14

**Comment by liangxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.10-amd64-nightly-4.10-upgrade-from-stable-4.9-gcp-ipi-workers-rhel8-p2-f14

**Comment by JianLi-RH** [comment]
> > will this PR fix https://issues.redhat.com/browse/OCPQE-14239?
> 
> Not exactly. It will fix part of the issue in OCPQE-14239. As for OCPQE-14239, @dis016 has submitted a PR (Thank you) which is stuck on issue [#37250 (comment)](https://github.com/openshift/release/pull/37250#issuecomment-1491220810). Will you take over it?

ok, will do

**Comment by shellyyang1989** [comment]
The [job](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/38103/rehearse-38103-periodic-ci-openshift-openshift-tests-private-release-4.10-amd64-nightly-4.10-upgrade-from-stable-4.9-gcp-ipi-workers-rhel8-p2-f14/1644242378464169984) failed due to permission issue.

Updating RHEL node repo
environment: line 97: repo.yaml: Permission denied

Does default directory in verification-tests image have write permission? Should we change a directory for RHEL upgrade?

**Comment by liangxia** [comment]
> Does default directory in verification-tests image have write permission? Should we change a directory for RHEL upgrade?

Fix the permission issue in https://github.com/openshift/release/pull/38176

**Comment by liangxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.10-amd64-nightly-4.10-upgrade-from-stable-4.9-gcp-ipi-workers-rhel8-p2-f14

**Comment by shellyyang1989** [comment]
The [test](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/38103/rehearse-38103-periodic-ci-openshift-openshift-tests-private-release-4.11-amd64-nightly-4.11-upgrade-from-stable-4.10-aws-ipi-proxy-workers-rhel8-p2-f14/1644242222515752960) failed as 

```
ERROR! the playbook: playbooks/upgrade.yml could not be found
```

The `openshift-ansible` is extracted in `/usr/share/ansible/`. Seems like we need to change `playbooks/upgrade.yml` to `/usr/share/ansible/playbooks/upgrade.yml`. Fixing in https://github.com/openshift/release/pull/38186

**Comment by liangxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.10-amd64-nightly-4.10-upgrade-from-stable-4.9-gcp-ipi-workers-rhel8-p2-f14

**Comment by liangxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-4.13-upgrade-from-stable-4.12-azure-ipi-workers-rhel8-p2-f14

**Comment by liangxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.11-amd64-nightly-4.11-upgrade-from-stable-4.10-aws-ipi-proxy-workers-rhel8-p2-f14

**Comment by liangxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.10-amd64-nightly-4.10-upgrade-from-stable-4.9-gcp-ipi-workers-rhel8-p2-f14

**Comment by liangxia** [comment]
The test failed with error,
```
Failed to create temporary directory. In some cases, you may have been able to authenticate and did not have permissions on the target directory. Consider changing the remote tmp path in ansible.cfg to a path rooted in \"/tmp\", for more error information use -vvv. Failed command was: ( umask 77 && mkdir -p \"` echo /.ansible/tmp `\"&& mkdir \"` echo /.ansible/tmp/ansible-tmp-1681142912.5367908-777-196543933656744 `\" && echo ansible-tmp-1681142912.5367908-777-196543933656744=\"` echo /.ansible/tmp/ansible-tmp-1681142912.5367908-777-196543933656744 `\" ), exited with result 1
```
Workaround in https://github.com/openshift/verification-tests/pull/3294
And the fix in https://github.com/openshift/openshift-ansible/pull/12446

**Comment by liangxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.10-amd64-nightly-4.10-upgrade-from-stable-4.9-gcp-ipi-workers-rhel8-p2-f14

**Comment by liangxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.11-amd64-nightly-4.11-upgrade-from-stable-4.10-aws-ipi-proxy-workers-rhel8-p2-f14

**Comment by liangxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.10-amd64-nightly-4.10-upgrade-from-stable-4.9-gcp-ipi-workers-rhel8-p2-f14

**Comment by liangxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.10-amd64-nightly-4.10-upgrade-from-stable-4.9-gcp-ipi-workers-rhel8-p2-f14

**Comment by liangxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.11-amd64-nightly-4.11-upgrade-from-stable-4.10-aws-ipi-proxy-workers-rhel8-p2-f14

**Comment by liangxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.10-amd64-nightly-4.10-upgrade-from-stable-4.9-gcp-ipi-workers-rhel8-p2-f14

**Comment by liangxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.10-amd64-nightly-4.10-upgrade-from-stable-4.9-gcp-ipi-workers-rhel8-p2-f14

**Comment by liangxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.11-amd64-nightly-4.11-upgrade-from-stable-4.10-aws-ipi-proxy-workers-rhel8-p2-f14

**Comment by liangxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.11-amd64-nightly-4.11-upgrade-from-stable-4.10-aws-ipi-proxy-workers-rhel8-p2-f14

**Comment by liangxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.10-amd64-nightly-4.10-upgrade-from-stable-4.9-gcp-ipi-workers-rhel8-p2-f14

**Comment by liangxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.10-amd64-nightly-4.10-upgrade-from-stable-4.9-gcp-ipi-workers-rhel8-p2-f14

**Comment by liangxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.11-amd64-nightly-4.11-upgrade-from-stable-4.10-aws-ipi-proxy-workers-rhel8-p2-f14

**Comment by liangxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.11-amd64-nightly-4.11-upgrade-from-stable-4.10-aws-ipi-proxy-workers-rhel8-p2-f14

**Comment by liangxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.10-amd64-nightly-4.10-upgrade-from-stable-4.9-gcp-ipi-workers-rhel8-p2-f14

**Comment by liangxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.11-amd64-nightly-4.11-upgrade-from-stable-4.10-aws-ipi-proxy-workers-rhel8-p2-f14

**Comment by liangxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.10-amd64-nightly-4.10-upgrade-from-stable-4.9-gcp-ipi-workers-rhel8-p2-f14

**Comment by liangxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.11-amd64-nightly-4.11-upgrade-from-stable-4.10-aws-ipi-proxy-workers-rhel8-p2-f14

**Comment by shellyyang1989** [comment]
Thank you Liang for your great support!

/lgtm

**Comment by liangxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.10-amd64-nightly-4.10-upgrade-from-stable-4.9-gcp-ipi-workers-rhel8-p2-f14

**Comment by liangxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.11-amd64-nightly-4.11-upgrade-from-stable-4.10-aws-ipi-proxy-workers-rhel8-p2-f14

**Comment by liangxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-4.13-upgrade-from-stable-4.12-azure-ipi-workers-rhel8-p2-f14

**Comment by liangxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-stable-4.13-upgrade-from-stable-4.12-azure-upi-p3-f28

**Comment by liangxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-arm64-nightly-4.13-upgrade-from-stable-4.13-azure-upi-p3-f28

**Pull request merged**